packages/aws-cdk/lib/api/cxapp/cloud-executable.ts

```diff
@@ -87,7 +87,11 @@ export class CloudExecutable {
         if (tryLookup) {
           debug('Some context information is missing. Fetching...');
 
-          await contextproviders.provideContextValues(assembly.manifest.missing, this.props.configuration.context, this.props.sdkProvider);
+          await contextproviders.provideContextValues(
+            assembly.manifest.missing,
+            this.props.configuration.context,
+            this.props.sdkProvider,
+            assembly.stacks[0].environment);
```

Comment:
This is a dangerous assumption to make, and not necessarily true.

There could be multiple stacks with multiple environments in the same cloud assembly, each having a "missing context" that needs a lookup.

The answer is we don't have a nice neat object that has what you need already lying around--however, every `MissingContext` object that *could* have an `assumeRoleArn` MUST ALSO have `account` and `region` fields, so you can construct the target `Environment` on the fly from those two fields.